File: RepoTransBench/target_projects/Python/Go/asciimoo_drawille/public_tests/test_public_speed_test.go
package public_tests

import (
	"testing"
)

import drawille "github.com/example/asciimoo_drawille"

func TestCanvasSpeedPublic(t *testing.T) {
	canvas := drawille.NewCanvas()
	for y := 0; y < 80; y += 2 {
		canvas.Set(15, y)
	}
	buf := canvas.Frame()
	hasBraille := false
	for _, c := range buf {
		if c >= 0x2800 {
			hasBraille = true
			break
		}
	}
	if !hasBraille {
		t.Errorf("Canvas.Frame() should contain at least one braille character")
	}
	canvas.Clear()
	buf2 := canvas.Frame()
	for _, c := range buf2 {
		if c >= 0x2800 {
			t.Errorf("After Canvas.Clear(), frame should not contain any braille char")
		}
	}
}

File: RepoTransBench/target_projects/Python/Rust/asciimoo_drawille/public_tests/test_public_speed_test.rs
// Translated from public_tests/test_public_speed_test.py

use asciimoo_drawille::Canvas;

#[test]
fn test_canvas_speed_public() {
    let c = Canvas::new();
    for y in (0..80).step_by(2) {
        c.set(15, y);
    }
    let buf = c.frame_no_args();

    // Buffer should not be empty and should contain braille chars
    assert!(buf.chars().any(|ch| ch as u32 >= 0x2800));
    c.clear();
    let buf2 = c.frame_no_args();
    assert!(!buf2.chars().any(|ch| ch as u32 >= 0x2800));
}